python/python/lancedb/query.py

```diff
@@ -1378,6 +1379,10 @@ def __init__(self, inner: Union[LanceQuery | LanceVectorQuery]):
         """
         self._inner = inner
 
+    def rerank(self, reranker=Reranker) -> AsyncQuery:
+        self._reranker = reranker
+        return self._inner.rerank(reranker)
+
```

Comment:
I'm not sure this method is needed. Since reranking only makes sense in the context of certain query types (vector, FTS or Hybrid search), we could probably refrain from adding reranker to the base async query.